libsolidity/formal/BMC.h

```diff
@@ -188,14 +192,17 @@ class BMC: public SMTEncoder
 		std::string const& _additionalValueName = "",
 		smtutil::Expression const* _additionalValue = nullptr
 	);
-	/// Checks that a boolean condition is not constant. Do not warn if the expression
-	/// is a literal constant.
-	void checkBooleanNotConstant(
-		Expression const& _condition,
-		smtutil::Expression const& _constraints,
-		smtutil::Expression const& _value,
-		std::vector<CallStackEntry> const& _callStack
-	);
+
+	struct ConstantExpressionCheckResult
+	{
+		smtutil::CheckResult canBeTrue;
+		smtutil::CheckResult canBeFalse;
+	};
+
+	/// Checks whether the given boolean condition is either true or always false under given constraints.
+	/// Returns the results from the solver for these two checks.
+	ConstantExpressionCheckResult
+	checkBooleanNotConstant(smtutil::Expression const& _constraints, smtutil::Expression const& _condition);
```

Comment:
The style with the return value on a separate line always looks confusing to me. I'd pretty much always prefer this:

```suggestion
	ConstantExpressionCheckResult checkBooleanNotConstant(
		smtutil::Expression const& _constraints,
		smtutil::Expression const& _condition
	);
```

Reply:
After having another look I agree that your version is more esthetically pleasing. I hope I got it right.